### PR TITLE
feat(eslint-plugin): Add new rule that preferes length checks

### PR DESCRIPTION
**Pull request by kresimir-coko**
There are 583 errors in `liferay-portal` when I run this rule. We probably want it to autofix because of the number of errors, I assume? @bryceosterhaus 

It can probably be improved, just code-wise, so setting it as a draft for now.

### CODE REVIEW

File: projects/eslint-plugin/rules/general/lib/rules/prefer-length-check.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+module.exports = {
+	create(context) {
+		return {
+			BinaryExpression(node) {
+				if (node.operator) {
+					if (
+						node.operator === '===' ||
+						node.operator === '<' ||
+						node.operator === '>'
```
Comment (bryceosterhaus) [inline projects/eslint-plugin/rules/general/lib/rules/prefer-length-check.js:16]: I think we would also need to check what the operator is comparing against. We only would want to check when explicitly checking against `0`


```js
// valid
if (foo.length > 5) {
}
```

**Review by bryceosterhaus — CHANGES_REQUESTED** [review]
@kresimir-coko yeah we will probably need an autofix so that we don't have to manually update every case

File: projects/eslint-plugin/rules/general/lib/rules/prefer-length-check.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const message = 'prefer using .length instead of .length === 0 or .length > 0';
+
+module.exports = {
+	create(context) {
+		return {
+			BinaryExpression(node) {
+				const {left, operator, right} = node;
+				const STRICT_EQUALITY = '===';
+				const MORE_THAN = '>';
+				const LENGTH = 'length';
```
Comment (bryceosterhaus) [inline projects/eslint-plugin/rules/general/lib/rules/prefer-length-check.js:15]: These can be moved up and out of this expression, that we only create these variables once.

If we leave them here, we essentially are creating these variables _everytime_ eslint crawls the AST and sees a BinaryExpression

File: projects/eslint-plugin/rules/general/tests/lib/rules/prefer-length-check.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+};
+
+const ruleTester = new MultiTester(parserOptions);
+
+const message = 'prefer using .length instead of .length === 0 or .length > 0';
+
+ruleTester.run('prefer-length-check', rule, {
+	invalid: [
+		{
+			code: `if (foo.length === 0) {}`,
```
Comment (bryceosterhaus) [inline projects/eslint-plugin/rules/general/tests/lib/rules/prefer-length-check.js:23]: We additionally should test against `if (0 === foo.length) {}`

File: projects/eslint-plugin/rules/general/tests/lib/rules/prefer-length-check.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+			errors: [
+				{
+					message,
+					type: 'BinaryExpression',
+				},
+			],
+			output: `if (!foo.length) {}`,
+		},
+		{
+			code: `if (foo.length > 0) {}`,
```
Comment (bryceosterhaus) [inline projects/eslint-plugin/rules/general/tests/lib/rules/prefer-length-check.js:33]: Same as above, we should also check for if the arguments are flipped `0 < foo.length`

**Review by bryceosterhaus — CHANGES_REQUESTED** [review]

File: projects/eslint-plugin/rules/general/lib/rules/prefer-length-check.js (hunk 2 above)
Comment (kresimir-coko) [reply projects/eslint-plugin/rules/general/lib/rules/prefer-length-check.js:15]: This one was a test for you, and you passed

**Comment by bryceosterhaus** [comment]
LGTM!

**Pull request merged**